Doc/library/sqlite3.rst

```diff
@@ -2614,6 +2590,49 @@ With some adjustments, the above recipe can be adapted to use a
 instead of a :class:`~collections.namedtuple`.
 
 
+.. _sqlite3-howto-encoding:
+
+How to handle non-UTF-8 text encodings
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :mod:`!sqlite3` adapts SQLite values with the ``TEXT`` data type
+using :class:`str`.
+This works well for UTF-8 encoded text, but it will fail for other encodings
+and invalid UTF-8.
+You can use a custom :attr:`~Connection.text_factory` to handle such cases.
+
+Because of SQLite's `flexible typing`_, it is not uncommon to encounter table
+columns with the ``TEXT`` data type, containing non-UTF-8 encodings,
+or even arbitrary data.
+To demonstrate, let's assume we've got a database with ISO-8859-2 (Latin 2)
+encoded text, for example a table of Czech-English dictionary entries.
+Assuming we now have a :class:`Connection` instance :py:data:`!con`
+connected to this database,
+we can decode the Latin 2 encoded text using this :attr:`~Connection.text_factory`:
+
+.. testcode::
+
+   con.text_factory = lambda data: str(data, encoding="latin2")
+
+For invalid UTF-8 or arbitrary data in stored in ``TEXT`` table columns,
+you can use the following technique, borrowed from the :ref:`unicode-howto`:
+
+.. testcode::
+
+   con.text_factory = lambda data: str(data, errors="surrogateescape")
+
+.. note::
+
+   Strings containing surrogate escapes and must be treated with care.
+   For example, you cannot pass them back to SQLite,
+   since the :mod:`!sqlite3` module API does not support strings
+   containing surrogate escape codes.
```

Comment:
The wording of this sentence is a bit off.  You can talk about surrogates if you are talking about the characters themselves.  `surrogateescape` is an error handler that converts surrogates to Unicode characters in a different range, and you also have escape sequences (`'\uXXXX'`) that represent surrogates.

Does "pass them back" imply that you can get them out from the DB?

Reply:
Hm, yes this is certainly a bit off. The point we want to highlight (IMO) is that strings containing surrogate escape sequences will not be accepted by the `sqlite3` APIs. For example, `Cursor.execute` will not accept strings with surrogate escape sequences neither as the *sql* string, nor as a part of the *parameters* sequence.

Reply:
Let's unblock this PR by removing this note; we can add a note in a follow-up PR.